packages/editor/src/epics/contractConfiguration/updateContractConfig.epic.ts

```diff
@@ -0,0 +1,53 @@
+import { ofType } from 'redux-observable';
+import { AnyAction } from 'redux';
+import { compilerActions, panesActions, contractConfigActions } from '../../actions';
+import { withLatestFrom, switchMap, catchError } from 'rxjs/operators';
+import { IProjectItem, ProjectItemTypes } from '../../models';
+import { findItemByPath } from '../../reducers/explorerLib';
+import { empty, of } from 'rxjs';
+
+export const updateContractConfig = (action$: AnyAction, state$: any) => action$.pipe(
+    ofType(compilerActions.HANDLE_COMPILE_OUTPUT),
+    withLatestFrom(state$),
+    switchMap(([action, state]) => {
+        console.log(state$);
+        const firstKey = Object.keys(action.data.contracts)[0];
```

Comment:
This will fail if there is an error while compiling the contract and the action doesn't return any contracts. Please add additional check if there are any errors in `action.data.errors`

Reply:
How do I handle the check? Do I just console log a message in the if check?

Reply:
imho I would just silently return empty as it's not really relevant for user and it doesn't need to update the args when the compilation doesn't go through